### ISSUE TITLE
Incomplete const lookup scope

### ISSUE THREAD

**Issue by mchf** [YorickPeterse/ruby-lint#191]
I'm having this 2 files "project". Both files are located in one directory.

* file 'const_container.rb'
```
class ConstContainer
  STRING_CONST = 'String from ConstContainer'.freeze
end
```
* file 'const_container_user.rb'
```
require_relative 'const_container'

puts 'class ConstContainer carries these constants:'
puts "* a string: #{ConstContainer::STRING_CONST}"
```

When running ```ruby const_container_user.rb``` then expected output is produced:
```
class ConstContainer carries these constants:
* a string: String from ConstContainer
```

however, when checked using ```ruby-lint *.rb``` I get:
```
const_container.rb: warning: line 2, column 3: unused constant STRING_CONST
const_container_user.rb: error: line 4, column 21: undefined constant ConstContainer::STRING_CONST
const_container_user.rb: error: line 4, column 21: undefined constant ConstContainer
```

All those reports are false positives as
* ```STRING_CONST``` constant is used
* class ```ConstContainer``` is defined

**Comment by YorickPeterse** [comment]
Moving the Ruby files into the `lib` or `app` directories should solve this problem as those are the directories ruby-lint searches in for external Ruby files. See http://code.yorickpeterse.com/ruby-lint/latest/file.configuration.html#directories for more information.

ruby-lint does not follow `require` or `require_relative` calls, there are various issues/comments discussing why this isn't the case. See https://github.com/YorickPeterse/ruby-lint/pull/179#issuecomment-226502392 for a recent list.
